quinn-proto/src/connection/mod.rs

```diff
@@ -865,13 +865,13 @@ where
         let ack_eliciting = builder.ack_eliciting;
         let exact_number = builder.exact_number;
         let space_id = builder.space;
-        let (size, _) = self.finish_packet(builder, buffer);
+        let (size, padded) = self.finish_packet(builder, buffer);
         let sent = match sent {
             Some(sent) => sent,
             None => return,
         };
 
-        let size = match sent.padding || ack_eliciting {
+        let size = match padded || ack_eliciting {
```

Comment:
Good catch! `sent.padding` doesn't capture the cases where we pad for close frames and some other edge-cases